### PR TITLE
Blaze: only display UI to admins.

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

* Let's avoid displaying the Blaze UI to non-admins. Only site owners should be able to promote content on their site.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* peeHDf-zm-p2

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Start with a public site that's connected to WordPress.com, with 2 **connected** users. One of those users should be the admin, the other an editor.
* Go to Posts menu and move your mouse over published posts;
    * When logged in as the editor, you should not see any link.
    * When logged in as the admin, you should see the link.

### CODE REVIEW

**Comment by kraftbj** [comment]
The change looks fine, but needs test updates at least.

**Comment by oskosk** [comment]
@kraftbj how do I run locally the tests that are failing here?

**Comment by kraftbj** [comment]
> @kraftbj how do I run locally the tests that are failing here?

Using the fancy tooling: `jetpack test packages/blaze php`
or
cd into `packages/blaze/` and run `vendor/bin/phpunit`

**Comment by kraftbj** [comment]
My _guess_ is that we'd need the unit test to mock that the test is an user with `manage_options` caps, but didn't look very close.

**Comment by oskosk** [comment]
@kraftbj unfortunately running those commands locally I don't get the failures. That's why I asked

**Comment by kraftbj** [comment]
> @kraftbj unfortunately running those commands locally I don't get the failures. That's why I asked

Interesting. I do get those failures locally.

**Comment by kraftbj** [comment]
Any chance you had pulled down the branch before Jer's force-push and pulling it now doesn't update without forcing it?

File: projects/packages/blaze/src/class-blaze.php [hunk 1]
```diff
@@ -121,6 +121,11 @@ public static function should_initialize() {
 		$connection        = new Jetpack_Connection();
 		$site_id           = Jetpack_Connection::get_site_id();
 
+		// Only admins should be able to Blaze posts on a site.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return false;
```
Comment (dereksmart) [inline projects/packages/blaze/src/class-blaze.php:126]: In order to not override the filter: 
```suggestion
			$should_initialize = false; 
```

**Comment by kraftbj** [comment]
@oskosk I pushed some code that resolved the test failures for me locally.

**Comment by oskosk** [comment]
> Any chance you had pulled down the branch before Jer's force-push, and pulling it now doesn't update without forcing it?

That was it. I just force-pulled, checked out to a commit behind yours, and I do see the failures locally. Thanks!

File: projects/packages/blaze/src/class-blaze.php (hunk 1 above)
Comment (jeherve) [reply projects/packages/blaze/src/class-blaze.php:126]: That's what I had done at first, but I didn't want to run the other checks for a role where we're sure we do not want to show the UI.

**Comment by jeherve** [comment]
Thank you all for taking a look, and making changes to the tests. I made a few more changes to them to add coverage for that new condition.

**Review by kraftbj — APPROVED** [review]
LGTM. I'm working on the Wordbless side and will clean this up via a later PR to only create new users once.

File: projects/packages/blaze/src/class-blaze.php (hunk 1 above)
Comment (dereksmart) [reply projects/packages/blaze/src/class-blaze.php:126]: ok! I'm out of the loop, why are we so sure?

Comment (jeherve) [reply projects/packages/blaze/src/class-blaze.php:126]: We talked about adding the admin condition in peeHDf-zm-p2

**Pull request merged**